### PR TITLE
Add logtail for disttae

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3931
## What this PR does / why we need it:

### CODE REVIEW

File: pkg/vm/engine/disttae/db.go [hunk 1]
```diff
@@ -19,20 +19,52 @@ import (
...
+	return db
 }
 
 func (db *DB) Update(ctx context.Context, dnList []DNStore,
 	databaseId, tableId uint64, ts timestamp.Timestamp) error {
+	op, err := db.cli.New()
+	if err != nil {
+		return err
+	}
+	db.Lock()
```
Comment (fengttt) [inline pkg/vm/engine/disttae/db.go:45]: get an anonymous function and call defer?

File: pkg/vm/engine/disttae/db.go [hunk 2]
```diff
@@ -19,20 +19,52 @@ import (
...
+	parts, ok := db.tables[[2]uint64{databaseId, tableId}]
+	if !ok { // create a new table
+		parts = make(Partitions, len(db.dnMap))
+		for i := range parts {
+			parts[i] = new(Partition)
+		}
+		db.tables[[2]uint64{databaseId, tableId}] = parts
+	}
+	db.Unlock()
+	for _, dn := range dnList {
```
Comment (fengttt) [inline pkg/vm/engine/disttae/db.go:55]: You pull all dns?   If possible, we want to read from primarykey/partitionkey, decide which dn, and pull one.
It is fine if leave this as a future optimization.

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/db.go:55]: No, dnlist is the parameter, which dn is calculated for this pull

File: pkg/vm/engine/disttae/db.go (hunk 1 above)
Comment (nnsgmsone) [reply pkg/vm/engine/disttae/db.go:45]: yes

**Review by daviszhen — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by yingfeng — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**